problems/cipher.js

```diff
@@ -0,0 +1,259 @@
+
+```
+
+const Cipher = function(input) {
+    this.key = input;
+    this.alphabet = 'abcdefghijklmnopqrstuvwxyz';
+};
+
+/**
+ * The opposite of indexToChar.
+ * Takes in a character and an alphabet string. Returns the alphaIndex of the
+ * character. For instance, for the regular alphabet:
+ * char = a returns 0
+ * char = b returns 1
+ * char = z returns 25
+ * @param  {str} char     a single character to find the index of
+ * @param  {str} alphabet an alphabet string to match the character against to find the index
+ * @return {int}          an integer representing the index of the character
+ */
+const _charToAlphaIndex = (char, alphabet) => {
+    let alphaIndex = null;
+
+    while (alphaIndex === null) {
+        alphabet.split('').forEach((alphaChar) => {
+            if (alphaChar === char) {
+                alphaIndex = alphabet.indexOf(alphaChar);
+            }
+        });
+    };
+  return alphaIndex;
+};
+
+/**
+ * The opposite of charToAlphaIndex.
+ * Takes in a character index and an alphabet string. Returns the character at
+ * the given index using the alphabet string. For instance, for the regular alphabet:
+ * index = 0 returns 'a'
+ * index = 25 returns 'z'
+ * index = 26 returns 'a'
+ * We use % to convert the index to its corresponding index when wrapped around the alphabet.
+ * For instance, an index of 40 % 26 = 14, so we use the new index of 14 to find the correct
+ * character.
+ * @param  {int} index    integer representing the index of the character
+ * @param  {str} alphabet an alphabet string to match the index of the character against
+ * @return {str}          a single character at that position in the alphabet
+ */
+const _indexToChar = (index, alphabet) => {
+    // If we're given a negative index, we need to add the alphabet length
+    // to convert it to its positive match
+    if (index < 0) {
+        index = index + alphabet.length;
+    }
+    // We need to use % to make sure the index will fit on the alphabet
+    // e.g., if we're given an index of 40, that doesn't fit on the alphabet.
+    const convertedIndex = index % alphabet.length;
+
+    return alphabet.charAt(convertedIndex);
+};
+
+/**
+ * Validates a given key to ensure that the key is long enough to be able to
+ * properly encode or decode the stringToConvert. If the key is not long enough,
+ * this method will repeat the key characters until the key becomes long enough.
+ * For instance, given a key of 'ab' and a stringToConvert of 'sahar', the key will
+ * return as 'ababa'.
+ * // TO DO: Add more validation for non-alpha keys
+ * @param  {str} key            the key provided by the user
+ * @param  {str} stringToConvert the string we're taking action on to convert later
+ * @return {str}                the validated key
+ */
+const _validateKey = (key, stringToConvert) => {
+    if (key.length != stringToConvert.length) {
+        index = -1;
+        while (key.length != stringToConvert.length) {
```

Comment:
So we have a while loop here. How else can we accomplish this and not use while loop? On top of my head there are 2 things we might be able todo: 

1. Using a simple math, figure out how many time we need to concat the key as a whole, and then how many partial last concat. That way you can use for loop with one additional line at the end.
2. Also another approach would be instead of having a key that matches the length of the string we are trying to cipher, we can use the shorter key in place by just circling back to the first key-char as you are ciphering the stings, thus removing the need to this logic.

Reply:
Still working on this part.